compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1203,11 +1206,36 @@ impl<'a> Parser<'a> {
                 if !this.recover_nested_adt_item(kw::Enum)? {
                     return Ok((None, TrailingToken::None));
                 }
+                let enum_field_start_span = this.token.span;
                 let ident = this.parse_field_ident("enum", vlo)?;
+                if this.token.kind == token::Colon {
+                    let snapshot = this.clone();
+                    this.bump();
+                    match this.parse_ty() {
+                        Ok(_) => {
+                            let mut err = this.struct_span_err(
+                                enum_field_start_span.to(this.prev_token.span),
+                                "the enum cannot have a struct field declaration",
```

Comment:
Diagnostics in parser largely follow a “expected ...” style of messages. The suggested wording would likely be more consistent with the rest of the diagnostics coming from the parser.

```suggestion
                                "expected an `enum` variant, found a field",
```